### PR TITLE
Correctly load/initialize classes when using ConstantPool

### PR DESCRIPTION
**Pull request by gacholio**
- do not load classes in the "ifLoaded" calls
- never intitialize classes in the natives
- do not update the constant pool when querying from the natives

Fixes: #12016

[ci skip]

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

**Comment by gacholio** [comment]
jenkins test sanity,extended zlinux jdk11

**Comment by gacholio** [comment]
jenkins compile win jdk8

File: runtime/vm/resolvesupport.cpp [hunk 1]
```diff
@@ -2091,6 +2101,8 @@ resolveOpenJDKInvokeHandle(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cp
 
 	Trc_VM_resolveInvokeHandle_Entry(vmThread, ramCP, cpIndex, resolveFlags);
 
+	Assert_VM_true(J9_ARE_NO_BITS_SET(resolveFlags, J9_RESOLVE_FLAG_NO_CP_UPDATE));
```
Comment (tajila) [inline runtime/vm/resolvesupport.cpp:2104]: Why is this specific to `OpenJDKInvokeHandle`? shouldn't it apply to `resolveInvokeDynamic` and `resolveConstantDynamic` as well?

Comment (gacholio) [reply runtime/vm/resolvesupport.cpp:2104]: None of these are currently used by the `ConstantPool` code. `resolveInvokeDynamic` and `resolveConstantDynamic` already have a similar assertion. The OpenJDK implementation code is using CAS to update the cache, and potentially returning the old contents, so I thought it best to not to unnecessarily change the code. The assertion will prevent bugs creeping in in the future if this is eventually called differently.

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
@gacholio can you please open another PR for the 0.26 stream

**Comment by gacholio** [comment]
https://github.com/eclipse/openj9/pull/12164

**Pull request merged**